addons/website/tests/test_res_users.py

```diff
@@ -102,3 +102,11 @@ def test_multi_website_multi_company(self):
 
         self._create_and_check_portal_user(False, company_1, company_2, website_1, website_2)
         self._create_and_check_portal_user(True, company_1, company_2, website_1, website_2)
+
+    def test_archive_company_linked_to_website(self):
+        company = self.env['res.company'].create({'name': 'Company'})
+        self.env['website'].create({'name': "Website 1", 'company_id': company.id})
+
+        # The company cannot be archived because it has a website linked to it
+        with self.assertRaisesRegex(ValidationError, 'The company Company cannot be archived'):
```

Comment:
```suggestion
        with self.assertRaises(ValidationError, 'The company Company cannot be archived'):
```
What's the difference? It's actually the first time I see `assertRaisesRegex`, is there any reason to use it instead of `assertRaises`?

Reply:
I must admit that since it was my first time creating a Python test, I drew significant inspiration from `odoo/addons/base/tests/test_res_company.py`
```# The user cannot be unarchived because its default company is archived
with self.assertRaisesRegex(ValidationError, 'Company foo is not in the allowed companies'):
       user.action_unarchive()
 ```
 But indeed you're right, it should be easier with assertRaises.

Reply:
@rdeodoo  With further investigation, it seems to be better with assertRaisesRegex so we can check the content of the error text field, and not only the Error Type, that can be another ValidationError. Tell me if I'm wrong, but browsing the code base, I didn't see a self.assertRaises that fits my case.

Reply:
I don't think it's worth ensuring the message is the correct one, if we ensure the operation doesn't go through, it seems more than enough to me.
Especially since it means copying the exception message in the test, having to keep them sync'd forever.

But why not.. Feel free to use what you prefer here, personally I'd just go with 
```py
with self.assertRaises(ValidationError):
```